tests/server/jobs/test_job.py

```diff
@@ -208,48 +207,80 @@ def test_job_resume_on_new_job_runner(monkeypatch, tmp_path):
 
     backend_store_uri = f"sqlite:///{db_tmp_path / 'mlflow.db'!s}"
 
-    with _setup_job_runner(1, monkeypatch, runner1_tmp_path, backend_store_uri) as job_runner_proc:
+    with _setup_job_runner(monkeypatch, runner1_tmp_path, backend_store_uri) as job_runner_proc:
         job1_id = submit_job(basic_job_fun, {"x": 3, "y": 4, "sleep_secs": 0}).job_id
-        job2_id = submit_job(basic_job_fun, {"x": 5, "y": 6, "sleep_secs": 10}).job_id
+        job2_id = submit_job(basic_job_fun, {"x": 5, "y": 6, "sleep_secs": 2}).job_id
         job3_id = submit_job(basic_job_fun, {"x": 7, "y": 8, "sleep_secs": 0}).job_id
-        time.sleep(1)
-
-    # ensure the job runner process is killed.
-    job_runner_proc.wait()
+        wait_job_finalize(job1_id)
 
-    with _setup_job_runner(1, monkeypatch, runner2_tmp_path, backend_store_uri):
         # assert that job1 has done, job2 is running, and job3 is pending.
         assert_job_result(job1_id, JobStatus.SUCCEEDED, 7)
         assert_job_result(job2_id, JobStatus.RUNNING, None)
         assert_job_result(job3_id, JobStatus.PENDING, None)
-        time.sleep(10)
+
+    # ensure the job runner process is killed.
+    job_runner_proc.wait()
+
+    with _setup_job_runner(monkeypatch, runner2_tmp_path, backend_store_uri):
+        wait_job_finalize(job2_id)
+        wait_job_finalize(job3_id)
         # assert all jobs are done.
         assert_job_result(job1_id, JobStatus.SUCCEEDED, 7)
         assert_job_result(job2_id, JobStatus.SUCCEEDED, 11)
         assert_job_result(job3_id, JobStatus.SUCCEEDED, 15)
 
 
+@job_function(max_workers=2)
+def job_fun_parallelism2(x, y, sleep_secs=0):
+    if sleep_secs > 0:
+        time.sleep(sleep_secs)
+    return x + y
+
+
+@job_function(max_workers=3)
+def job_fun_parallelism3(x, y, sleep_secs=0):
+    if sleep_secs > 0:
+        time.sleep(sleep_secs)
+    return x + y
+
+
 def test_job_queue_parallelism(monkeypatch, tmp_path):
-    # test job queue parallelism=2 and each job consumes 2 seconds.
-    with _setup_job_runner(2, monkeypatch, tmp_path):
-        job_ids = [
-            submit_job(basic_job_fun, {"x": x, "y": 1, "sleep_secs": 3}).job_id for x in range(4)
+    with _setup_job_runner(monkeypatch, tmp_path):
+        # warm up:
+        # The first job submission of each job function triggers setting up
+        # the corresponding huey consumer process.
+        job1_id = submit_job(job_fun_parallelism2, {"x": 1, "y": 1, "sleep_secs": 0}).job_id
+        job2_id = submit_job(job_fun_parallelism3, {"x": 1, "y": 1, "sleep_secs": 0}).job_id
+        wait_job_finalize(job1_id)
+        wait_job_finalize(job2_id)
```

Comment:
can we add a `wait` method in the `Job` class?

Reply:
It is a bit weird to add the `Job.wait` API, because it needs a server URL param.